nvidia.go

```diff
@@ -0,0 +1,116 @@
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"golang.org/x/sys/unix"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+const ldconfPath = "/etc/ld.so.conf"
+
+func setupNvidiaDriver(containerRootfs string) (err error) {
+	bins, libs, err := getNvidiaFiles()
+	if err != nil {
+		return
+	}
+
+	// mount driver binaries
+	for _, binFile := range bins {
+		if err = mount(binFile, containerRootfs+binFile, "bind", unix.MS_BIND|unix.MS_REC, "rbind"); err != nil {
+			return
+		}
+	}
+
+	// mount driver libraries
+	libDirsMap := make(map[string]interface{})
+	for _, libFile := range libs {
+		if err = mount(libFile, containerRootfs+libFile, "bind", unix.MS_BIND|unix.MS_REC, "rbind"); err != nil {
+			return
+		}
+		libDir := filepath.Dir(libFile)
+		libDirsMap[libDir] = nil
+	}
+
+	// configure dynamic linker run-time bindings
+	ldArgs := []string{"-r", containerRootfs}
+	for dir := range libDirsMap {
+		ldArgs = append(ldArgs, containerRootfs+dir)
+	}
+	ldCmd := exec.Command("ldconfig", ldArgs...)
+	err = ldCmd.Run()
+	if err != nil {
+		return
+	}
+	// write libdirs to ld.so.conf
+	confFile, err := os.OpenFile(containerRootfs+ldconfPath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0644)
+	if err != nil {
+		return
+	}
+	defer confFile.Close()
+	for dir := range libDirsMap {
+		if _, err = fmt.Fprintln(confFile, dir); err != nil {
+			return
+		}
+	}
+
+	return nil
+}
+
+func getNvidiaFiles() (bins []string, libs []string, err error) {
+	// "-k": Load kernel modules
+	// "info": Report information about the driver and devices
+	infoCmd := exec.Command("nvidia-container-cli", "--load-kmods", "info")
```

Comment:
missing `-k` option? you mention it in the comments

Reply:
'-k' is just '--load-kmods', I have changed the code but forgot to change the comment.  I'll fix it.